boards/posix/native_posix/hw_models_top.c

```diff
@@ -190,4 +201,29 @@ void hwm_cleanup(void)
 	hw_irq_ctrl_cleanup();
 }
 
+/**
+ * Send a request to the HW model main loop to perform a exit.
+ *
+ * This function can be called from a background thread to perform a controlled
+ * exit in the main thread.
+ *
+ * @param exit_code Exit code to use
+ *
+ */
+void hwm_main_loop_exit(int exit_code)
+{
```

Comment:
Just a silly comment related to the previous comments: if you did not have the `exit_code` all this function body could be replaced with just `signaled_end = 1`.

Reply:
correct but in this way the thread can signal that an exit was performed due to a failure and the error can be propagated to the command line